main.qml

```diff
@@ -66,6 +66,8 @@ ApplicationWindow {
     property bool ctrlPressed: false
     property alias persistentSettings : persistentSettings
     property string accountsDir: !persistentSettings.portable ? moneroAccountsDir : persistentSettings.portableFolderName + "/wallets"
+    property bool firstHardwareWalletConnection: true
```

Comment:
This will always be true on every start. Is this intended?

Reply:
Yes. When `true`, this means the user has not yet seen the first message ("please connect device and unlock it"). After the user sees this message for the first time, this becomes `false` and the second message (troubleshooting) is displayed the next time connection fails.

Reply:
This is not a persistent variable. Every time the GUI gets restarted this will display.

Reply:
Sure, and that's intended. A better name for this variable would be `firstHardwareWalletConnectionOnThisSession`.

Reply:
Not sure it is necessary to display instructions during every session but I did not test the PR yet.

Reply:
The idea here is to display a nice message every time the user opens a wallet file of a hardware wallet, but trezor/ledger device is locked. 

I usually forget to unlock my trezor before clicking on the wallet in "open wallet" page, and then I get this error message "Could not connect to the device Trezor" that doesn't inform that the wallet is locked.

Ideally we should know before opening the wallet file that it is a hardware wallet file, and then we display this nice reminder that the user should unlock the hardware wallet first. However, it is not possible to see if the wallet file is from a hardware wallet before opening and decrypting it. 

So it works this way:
Open wallet file -> can't connect to hardware wallet -> display nice message to remind user to unlock it (on first try) -> `firstHardwareWalletConnection ` set to `false ` -> can't connect again -> display troubleshooting message

Reply:
Ok, I might have misunderstood it, like I said, I did not test it. So the message only shows up if you fail. This is fine then.